src/dishka/integrations/typer.py

```diff
@@ -0,0 +1,95 @@
+"""Integration for Typer https://typer.tiangolo.com"""
+
+__all__ = [
+    "FromDishka",
+    "inject",
+    "setup_dishka",
+]
+
+from collections.abc import Callable
+from inspect import Parameter
+from typing import Final, ParamSpec, TypeVar, cast, get_type_hints
+
+import click
+import typer
+
+from dishka import Container, FromDishka, Scope
+from dishka.dependency_source.make_context_var import from_context
+from dishka.provider import Provider
+from .base import is_dishka_injected, wrap_injection
+
+T = TypeVar("T")
+P = ParamSpec("P")
+CONTAINER_NAME: Final = "dishka_container"
+CONTAINER_NAME_REQ: Final = "dishka_container_req"
+
+
+def inject(func: Callable[P, T]) -> Callable[P, T]:
+    hints = get_type_hints(func)
+    context_hint = next(
+        (name for name, hint in hints.items() if hint is typer.Context),
+        None,
+    )
+    if context_hint is None:
+        additional_params = [
+            Parameter(
+                name="___dishka_context",
+                annotation=typer.Context,
+                kind=Parameter.KEYWORD_ONLY,
+            ),
+        ]
+    else:
+        additional_params = []
+
+    param_name = context_hint or "___dishka_context"
+
+    def get_container(_, p):
+        context: typer.Context = p[param_name]
+        container = context.meta[CONTAINER_NAME]
+
+        req_container = context.with_resource(
+            container({typer.Context: context}, scope=Scope.REQUEST),
```

Comment:
I do not think we should do it. I treat `typer` as well as `click` as framework which actually just runs app as a whole: inside click/typer handlers there can be other frameworks started (like fastapi, etc.)

So, from my point of view, expected behavior for typer handler is to have Scope.APP, not REQUEST. Let't keep the logic we have for click.

I see the problem now: you cannot pass `Context` because APP-scoped container is already created. My fault, let's rollback and ignore this requirement.

Reply:
Would you go back to `typer.Context` not being available to providers?

Reply:
Correct. We cannot pass it here, so let it be user's problem.